packages/core/src/interpreter.ts

```diff
@@ -954,16 +958,18 @@ export class Interpreter<
             return;
           }
 
-          let options: SpawnOptions | undefined;
-
           if (isMachine(source)) {
             source = resolvedData ? source.withContext(resolvedData) : source;
-            options = {
-              autoForward
-            };
           }
 
-          this.spawn(source, id, options);
+          const isLazyEntity =
+            isMachine(source) || isFunction(source) || isBehavior(source);
+          activity.deferred.start({
+            parent: this,
+            entity: isLazyEntity ? source : () => source
```

Comment:
so invoke entities are still resolved here, in the interpreter, they get suboptimal events/context arguments but at least the timing of things doesn't change here that much